### PR TITLE
Arduino code for IR sensor on Lateral Suspension

### PR DESCRIPTION
**Pull request by supermegho**
## Description
This is the initial Arduino code for counting the wheel revolutions on a lateral suspension wheel. A signal is sent to the BBB every time an input is received by one the IR pins.

 *Click up link* [](<link here>)

## Changes
- IR sensor pin takes an input
- Output is sent to BBB.

## Additional Info 
screenshots, commands to run etc.

### CODE REVIEW

File: arduino/irsensor.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    else {
+        digitalWrite (Output, HIGH);
+    }
+    if (statusSensor4 == 1){
+        digitalWrite (Output4, LOW);
+    }
+    else {
+        digitalWrite (Output4, HIGH);
+    }
```
Comment (MiltFra) [inline arduino/irsensor.cpp:56]: This is missing a closing bracket, no?

File: arduino/irsensor.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+void setup(){
+    // IRSensors are inputs
+    pinMode (IRSensor1, INPUT);
+    pinMode (IRSensor2, INPUT);
+    pinMode (IRSensor3, INPUT);
+    pinMode (IRSensor4, INPUT);
+    // Outputs
+    pinMode (Output1, OUTPUT);
+    pinMode (Output2, OUTPUT);
+    pinMode (Output3, OUTPUT);
```
Comment (MiltFra) [inline arduino/irsensor.cpp:21]: Formatting. `pinMode(Output3, OUTPUT);`

File: arduino/irsensor.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    pinMode (Output4, OUTPUT);
+}
+// Read IR Sensors + Output based on reading
+void loop(){
+    statusSensor1 = digitalRead (IRSensor1);
+    statusSensor2 = digitalRead (IRSensor2);
+    statusSensor3 = digitalRead (IRSensor3);
+    statusSensor4 = digitalRead (IRSensor4);
+
+    if (statusSensor1 == 1){
```
Comment (MiltFra) [inline arduino/irsensor.cpp:31]: No need to check the `== 1`. Just do `if (statusSensor1) { ... }`. Looks more intuitive, I think.

**Review by MiltFra — CHANGES_REQUESTED** [review]
This just passes on the signals, right? I'd like to see the counting implemented before we pass this on. @kshxtij

File: arduino/irsensor.cpp (hunk 1 above)
Comment (supermegho) [reply arduino/irsensor.cpp:56]: Yes, you are correct

**Comment by SnickeyX** [comment]
@kshxtij and I spent some time trying to get the arduino to work. Most things failed, the strings method works for now; so committing this while we keep experimenting.

**Comment by kshxtij** [comment]
Replaced by #120.

**Pull request closed without merging**